src/middlewares/validators/signup.validation.js

```diff
@@ -0,0 +1,96 @@
+import Joi from 'joi';
+import isAlphaNumeric from './is.alphanumeric';
+import Util from '../../helpers/util';
+
+const util = new Util();
+
+export default (req, res, next) => {
+  const {
+    firstname, lastname, email, password, username
+  } = req.body;
+
+  if (password && !isAlphaNumeric(password)) {
+    util.setError(400, 'Password should be Alphanumeric');
+    return util.send(res);
+  }
+
+  const schema = {
+    firstname: Joi.string()
+      .trim()
+      .regex(/^[a-zA-Z]+$/)
+      .min(1)
+      .max(20)
+      .required(),
+    lastname: Joi.string()
+      .trim()
+      .regex(/^[a-zA-Z]+$/)
+      .min(1)
+      .max(20)
+      .required(),
+    username: Joi.string()
+      .trim()
+      .min(1)
+      .max(20)
+      .required(),
+    password: Joi.string()
+      .alphanum()
+      .min(8)
+      .required(),
+    email: Joi.string()
+      .email({ minDomainAtoms: 2 })
+      .required()
+  };
+  const { error } = Joi.validate({
+    firstname, lastname, password, email, username
+  }, schema);
+
+  if (!error) return next();
+  const errorMessageFromJoi = error.details[0].message;
+  switch (errorMessageFromJoi) {
+    case '"password" is not allowed to be empty':
+      util.setError(400, 'No password was specified');
+      return util.send(res);
+    case '"password" is required':
+      util.setError(400, 'No password was specified');
+      return util.send(res);
+    case '"password" length must be at least 8 characters long':
+      util.setError(400, 'Password length must be at least 8 characters long');
+      return util.send(res);
+    case '"email" is not allowed to be empty':
+      util.setError(400, 'No email was specified');
+      return util.send(res);
+    case '"email" must be a valid email':
+      util.setError(400, 'Email is not valid');
+      return util.send(res);
+    case '"email" is required':
+      util.setError(400, 'No email was specified');
+      return util.send(res);
+    case '"firstname" is not allowed to be empty':
+      util.setError(400, 'No firstname was specified');
+      return util.send(res);
+    case '"firstname" is required':
+      util.setError(400, 'No firstname was specified');
+      return util.send(res);
```

Comment:
I would like to get the difference between case 'first name' is not allowed to be empty and 'first name' is required

Reply:
If the key firstname is missing in the request.body, joi error message will be firstname is required but if the key is there but is empty then joi error message will be firstname is not allowed to be empty.  I customized those JOI error messages to descriptive and polite error messages